templates/mesh-gateway-deployment.yaml

```diff
@@ -0,0 +1,162 @@
+    component: mesh-gateway
+spec:
+  replicas: {{ .Values.meshGateway.replicas }}
+  selector:
+    matchLabels:
+      app: {{ template "consul.name" . }}
+      chart: {{ template "consul.chart" . }}
+      release: {{ .Release.Name }}
+      component: mesh-gateway
+  template:
+    metadata:
+      labels:
+        app: {{ template "consul.name" . }}
+        chart: {{ template "consul.chart" . }}
+        release: {{ .Release.Name }}
+        component: mesh-gateway
+      annotations:
+        "consul.hashicorp.com/connect-inject": "false"
+        {{- if .Values.meshGateway.annotations }}
+          {{- tpl .Values.meshGateway.annotations . | nindent 8 }}
+        {{- end }}
+    spec:
+      {{- if .Values.meshGateway.affinity }}
+      affinity:
+        {{ tpl .Values.meshGateway.affinity . | nindent 8 | trim }}
+      {{- end }}
+      {{- if .Values.meshGateway.tolerations }}
+      tolerations:
+        {{ tpl .Values.meshGateway.tolerations . | nindent 8 | trim }}
+      {{- end }}
+      terminationGracePeriodSeconds: 10
+      serviceAccountName: {{ template "consul.fullname" . }}-mesh-gateway
+      volumes:
+      - name: consul-bin
+        emptyDir: {}
+      hostNetwork: {{ .Values.meshGateway.hostNetwork }}
+      dnsPolicy: {{ .Values.meshGateway.dnsPolicy }}
+      initContainers:
+      # We use the Envoy image as our base image so we use an init container to
+      # copy the Consul binary to a shared directory that can be used when
+      # starting Envoy.
+      - name: copy-consul-bin
+        image: {{ .Values.global.image | quote }}
+        command:
+        - cp
+        - /bin/consul
+        - /consul-bin/consul
+        volumeMounts:
+        - name: consul-bin
+          mountPath: /consul-bin
+      {{- if .Values.global.bootstrapACLs }}
+      # Wait for secret containing acl token to be ready.
+      # Doesn't do anything with it but when the main container starts we
+      # know that it's been created.
+      - name: mesh-gateway-acl-init
+        image: {{ .Values.global.imageK8S }}
+        command:
+          - "/bin/sh"
+          - "-ec"
+          - |
+            consul-k8s acl-init \
+              -secret-name="{{ .Release.Name }}-consul-mesh-gateway-acl-token" \
+              -k8s-namespace={{ .Release.Namespace }} \
+              -init-type="sync"
+      {{- end }}
+      containers:
+      - name: mesh-gateway
+        image: {{ .Values.meshGateway.imageEnvoy | quote }}
+        {{- if .Values.meshGateway.resources }}
+        resources:
+          {{ tpl .Values.meshGateway.resources . | nindent 10 | trim }}
+        {{- end }}
+        volumeMounts:
+        - name: consul-bin
+          mountPath: /consul-bin
+        env:
+        - name: HOST_IP
+          valueFrom:
+            fieldRef:
+              fieldPath: status.hostIP
+        - name: POD_IP
+          valueFrom:
+            fieldRef:
+              fieldPath: status.podIP
+        - name: NODE_NAME
```

Comment:
This should be optional, only if `.Values.meshGateway.wanAddress.useNodeName` is true.

Reply:
Is there a downside to having it here regardless? My thought was that it's less conditionals == less complexity.